packages/rabbitmq/src/amqp/connection.ts

```diff
@@ -343,17 +343,19 @@ export class AmqpConnection {
     originalHandlerName: string
   ): Promise<SubscriptionResult> {
     return new Promise((res) => {
-      this.selectManagedChannel(msgOptions?.queueOptions?.channel).addSetup(
-        async (channel) => {
+      this.selectManagedChannel(msgOptions?.queueOptions?.channel)
+        .addSetup(async (channel) => {
           const consumerTag = await this.setupSubscriberChannel<T>(
             handler,
             msgOptions,
             channel,
             originalHandlerName
           );
           res({ consumerTag });
-        }
-      );
+        })
+        .then((result: any) => {
+          res(result);
```

Comment:
I'm confused as to how we can resolve two values from the same promise. Doesn't the promise always get resolved before it can get to this new `then` continuation when we resolve the consumer tag on line 354?

Reply:
Hi @WonderPanda , the addSetup function actually either calls the callback or return undefined. We have handled the call back but didn't handle the returned value.

https://github.com/jwalton/node-amqp-connection-manager/blob/a1092b3a3a8e866080a1b400cd826a8edb739e3d/src/ChannelWrapper.ts#L228

This is the code snippet of the addSetup function. Please let me know if the returned value can be handled differently.

Reply:
This patch worked when I applied it locally using `patch-package`. However, I wonder if there is a way to rewrite it such that there is only one invocation of `res`, e.g.:

```typescript
    originalHandlerName: string
  ): Promise<SubscriptionResult> {
    return new Promise((res) => {
      let result;
      this.selectManagedChannel(msgOptions?.queueOptions?.channel).addSetup(
        async (channel) => {
      this.selectManagedChannel(msgOptions?.queueOptions?.channel)
        .addSetup(async (channel) => {
          const consumerTag = await this.setupSubscriberChannel<T>(
            handler,
            msgOptions,
            channel,
            originalHandlerName
          );
          result = { consumerTag };
        }
      );
        })
        .then(() => {
          res(result);
```

I'm not very familiar with the `promise-breaker` package and the `addSetup` function, so I am not sure if this is appropriate.